core/templates/dev/head/services/DebouncerService.ts

```diff
@@ -16,41 +16,48 @@
  * @fileoverview Service for debouncing function calls.
  */
 
-angular.module('oppia').factory('DebouncerService', [function() {
-  return {
-    // Returns a function that will not be triggered as long as it continues to
-    // be invoked. The function only gets executed after it stops being called
-    // for `wait` milliseconds.
-    debounce: function(func, millisecsToWait) {
-      var timeout;
-      var context = this;
-      var args = arguments;
-      var timestamp;
-      var result;
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
 
-      var later = function() {
-        var last = new Date().getTime() - timestamp;
-        if (last < millisecsToWait) {
-          timeout = setTimeout(later, millisecsToWait - last);
-        } else {
-          timeout = null;
-          result = func.apply(context, args);
-          if (!timeout) {
-            context = null;
-            args = null;
-          }
-        }
-      };
+@Injectable({
+  providedIn: 'root'
+})
+export class DebouncerService {
+  // Returns a function that will not be triggered as long as it continues to
+  // be invoked. The function only gets executed after it stops being called
+  // for `wait` milliseconds.
+  debounce(func: any, millisecsToWait: any) {
+    let timeout;
+    let context = this;
+    let args = arguments;
+    let timestamp;
+    let result;
 
-      return function() {
-        context = this;
-        args = arguments;
-        timestamp = new Date().getTime();
+    let later = () => {
+      let last = new Date().getTime() - timestamp;
```

Comment:
I doubt if this will work in Angular. Please confirm.

Reply:
@bansalnitish i think these should work just fine because both of the functions are inside the main debounce function and not member functions.

Reply:
Umm, How will angular know what's new Date() when we are not importing date object anywhere? (I'm not sure how it worked with AngularJS)

Reply:
Date isn't a Angular function, it's a javascript function. Hence, We don't need to import that.
@bansalnitish

Reply:
Okay cool.